lib/std/os/linux.zig

```diff
@@ -6209,8 +6224,13 @@ pub const io_uring_buf_reg = extern struct {
     ring_addr: u64,
     ring_entries: u32,
     bgid: u16,
-    pad: u16,
+    flags: u16,
     resv: [3]u64,
+
+    pub const FLAG = struct {
+        // Incremental buffer consummation.
+        pub const INC: u16 = 2;
+    };
```

Comment:
This should be a `packed struct(u16)`

Reply:
'Original' liburing definition is [here](https://github.com/axboe/liburing/blob/e1003e496e66f9b0ae06674869795edf772d5500/src/include/liburing/io_uring.h#L796).
I can put IOU_PBUF_RING_INC as pub const to be accessible as linux.IOU_PBUF_RING_INC but it is used only for this io_uring_buf_reg structure so I put it inside that struct. **FLAG is here just namespace** so there is no need for packing. 
By Zig convention struct which are just namespace should be in lowercase, but linux.zig is mostly copying names from C header files.

Reply:
It's for type safety, packed structs are preferred over manually ORing flags together.

Reply:
I don't understand this point... FLAG is a zero sized struct, used purely for namespacing. How could it be ORed?

Reply:
`IOU_PBUF_RING_MMAP | IOU_PBUF_RING_INC` is `.{ .MMAP = true, .INC = true }` in Zig land. I don't know why @ianic omitted the MMAP flag but they might well add more in the future.

https://github.com/torvalds/linux/blob/99fa936e8e4f117d62f229003c9799686f74cebc/io_uring/kbuf.c#L633-L634

Arguably we can lower the field names, it's fairly common: https://github.com/search?q=repo%3Aziglang%2Fzig+path%3A%2F%5Elib%5C%2Fstd%5C%2Fos%5C%2F%2F+%22packed+struct%28u%22&type=code - this only really makes sense when mapping something like `MAP_ANONYMOUS` to `std.os.linux.MAP.ANONYMOUS` (pun intended).

Specific suggestion to avoid ambiguity:

```suggestion
    pub const Flags = packed struct(u16) {
        /// IOU_PBUF_RING_MMAP
        mmap: bool,
        /// IOU_PBUF_RING_INC
        inc: bool,
        _: u14,
    };
```

Reply:
IOU_PBUF_RING_MMAP is in [liburing](https://github.com/axboe/liburing/blob/e1003e496e66f9b0ae06674869795edf772d5500/src/setup.c#L596) used only when compiling for \_\_hppa\_\_ (PA-RISC) architecture. That changes who is mmaping buffers in setup_buf_ring userspace or kernel. That architecture is dead more then a decade. 

In current Zig's implementation we are always mmaping in userspace, so I didn't include that flag. If used it will change kernel behavior and make it incompatible with Zig io_uring. 

Introduced [here](https://github.com/axboe/liburing/commit/9c6689848ebf79a5830258c6416101b86b05f35e) in liburing.